services/alchemy_language/v1.js

```diff
@@ -25,77 +28,114 @@ var FEATURES = {
 
   var service = cfenv.getAppEnv().getServiceCreds(/alchemy/i);
 
   if (service) {
-    apikey = service.apikey;
+    s_apikey = service.apikey;
   }
 
   RED.httpAdmin.get('/alchemy-feature-extract/vcap', function (req, res) {
     res.json(service ? {bound_service: true} : null);
   });
 
+
+  // This is the Alchemy Data Node
+
   function AlchemyFeatureExtractNode (config) {
     RED.nodes.createNode(this, config);
     var node = this;
 
     this.on('input', function (msg) {
       if (!msg.payload) {
+        this.status({fill:'red', shape:'ring', text:'missing payload'});          
         var message = 'Missing property: msg.payload';
         node.error(message, msg);
         return;
       }
 
-      apikey = apikey || this.credentials.apikey;
+      // If it is present the newly provided user entered key takes precedence over the existing one. 
+      apikey = s_apikey || this.credentials.apikey;
+      this.status({});  
 
       if (!apikey) {
+        this.status({fill:'red', shape:'ring', text:'missing credentials'});          
         var message = 'Missing Alchemy API service credentials';
         node.error(message, msg);
         return;
       }
 
-      var alchemy = new AlchemyAPI(apikey);
+      var alchemy_language = watson.alchemy_language( { api_key: apikey } );
+
+      // Check which features have been requested.
 
       var enabled_features = Object.keys(FEATURES).filter(function (feature) { 
         return config[feature]
       });
 
+
       if (!enabled_features.length) {
+        this.status({fill:'red', shape:'ring', text:'no features selected'});    
         var message = 'AlchemyAPI node must have at least one selected feature.';
         node.error(message, msg);
         return;
       }
 
-      alchemy.combined(msg.payload, enabled_features, msg.alchemy_options || {}, function (err, response) {
-        if (err || response.status === 'ERROR') { 
-          var message = 'Alchemy API request error: ' + (err ? err : response.statusInfo);
-          node.error(message, msg);
-          return;
-        }
+       // The watson node-SDK expects the features as a single string. 
+      var extract = "" ; //doc-sentiment";
+      enabled_features.forEach(function(entry){extract += (',' + entry)})     
+
+      //console.log("Will be looking for ", extract)
 
-        msg.features = {};
-        Object.keys(FEATURES).forEach(function (feature) { 
-          var answer_feature = FEATURES[feature];
-          msg.features[feature] = response[answer_feature];
-        });
+      var params = { text: msg.payload, extract: extract  };
+
+      alchemy_language.combined(params, function (err, response) {
+        if (err || response.status === 'ERROR') {
+          node.status({fill:'red', shape:'ring', text:'call to alchmeyapi language service failed'}); 
+          console.log('Error:', msg, err);
+          node.error(err, msg);
+        }
+        else {
+           msg.features = {};
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected space or tab after "//" in comment.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2557150214/issues/source?bid=3225115&fileBranchId=3292252#l122)